### ISSUE TITLE
Install as Service: /usr/local/bin/iptsd Permission Denied

### ISSUE THREAD

**Issue by swinzy** [quo/iptsd#14]
Hi all,
I have a Surface Laptop Studio with quo/ithc-linux installed to enable touch support.
Now I've built the iptsd trying to enable multi-touch support. It runs manually.
I'm trying to install it as a service, but the service won't start and here's the log:
```
× iptsd.service - Intel Precise Touch & Stylus Daemon
     Loaded: loaded (/usr/lib/systemd/system/iptsd.service; enabled; vendor preset: disabled)
     Active: failed (Result: exit-code) since Wed 2022-08-03 14:05:08 AEST; 3min 34s ago
       Docs: https://github.com/linux-surface/iptsd
    Process: 14167 ExecStart=/usr/local/bin/iptsd (code=exited, status=203/EXEC)
   Main PID: 14167 (code=exited, status=203/EXEC)
        CPU: 902us

Aug 03 14:05:08 crusher-studil systemd[1]: Started iptsd.service - Intel Precise Touch & Stylus Daemon.
Aug 03 14:05:08 crusher-studil systemd[14167]: iptsd.service: Failed to locate executable /usr/local/bin/iptsd: Permission denied
Aug 03 14:05:08 crusher-studil systemd[14167]: iptsd.service: Failed at step EXEC spawning /usr/local/bin/iptsd: Permission denied
Aug 03 14:05:08 crusher-studil systemd[1]: iptsd.service: Main process exited, code=exited, status=203/EXEC
Aug 03 14:05:08 crusher-studil systemd[1]: iptsd.service: Failed with result 'exit-code'.
```

By the way I've added a+rwx to /usr/local/bin/iptsd and also a+r to /dev/ithc.
So now even if I open a terminal and run /usr/local/bin/iptsd without sudo, it works perfectly with multi-touch enabled.

And here's the iptsd.service file (I had to add [Install] because I didn't have an iptsd service before and it won't enable or start without that):
```
[Unit]
Description=Intel Precise Touch & Stylus Daemon
Documentation=https://github.com/linux-surface/iptsd

[Service]
Type=simple
ExecStart=/usr/local/bin/iptsd

[Install]
WantedBy=default.target
```

Anyone has any idea what could be the reason?
Cheers
Stephen

**Comment by NP-chaonay** [comment]
1) /usr/local/bin/iptsd should be 0755 and /dev/ithc should be 440. (permission value used in chmod)
2) Are you using Ubuntu, I have tested on Ubuntu, if not then check if  SELinux or SELinux-alike app reports error when ipstd is executed.
3) set that file in 1) as 1) and try to execute service, oh before execute pls check your sevice file in 4)
4) when running ```nano /lib/systemd/system/iptsd.service``` you should only see this, no other thing else
![image](https://user-images.githubusercontent.com/17851493/182545394-cb6b1ec8-2ee6-4f42-8b4f-ed4f415f0513.png)
it doesnt have ```[Install]``` section because the service will be trigger to be excuted when udev detected ithc device (the evidence is two pict below)
![image](https://user-images.githubusercontent.com/17851493/182544620-afc643df-4b8b-4174-89d1-676313268339.png)
![image](https://user-images.githubusercontent.com/17851493/182544689-c5c31606-cae0-4048-81ad-a5383be68850.png)

**Comment by NP-chaonay** [comment]
So if your system is met one of these:
1) no systemd
2) service file is not what I specified
3) permission of iptsd binary and /dev/ithc is not same as I specified
4) your system not have udev or it is disabled

you may in trouble
if just 3) 2) you can fix it.
PS: Im using Ubuntu 22.04 LTS with quo iptsd on SLS

**Comment by NP-chaonay** [comment]
so if your service file is broken or not same as me then try to remove your manual service file

and put the text I have send and save in /lib/systemd/system/iptsd.service

and make sure that permission of iptsd and /dev/ithc is correct and then try to reboot

**Comment by swinzy** [comment]
@NP-chaonay Thanks for the reply.
At the beginning, when I use the service file without [Install] section it kept saying something like "cannot find iptsd.service".
This time I've changed to exactly what you posted there.
![image](https://user-images.githubusercontent.com/19246983/182597110-baaa50d9-55ad-4131-8a34-d32e40653cbd.png)

> /usr/local/bin/iptsd should be 0755 and /dev/ithc should be 440. (permission value used in chmod)

Done, those files are now in the exact permissions you mentioned above.

**Update: I've tried disabling SELinux temporarily with ```setenforce 0``` and it works now.**
The problem is, how should I make it work **with** SELinux?

~~I'm using Fedora. I've never dealt with SELinux but I surely can see something related to iptsd in the log file. I've attached it below:~~
~~Not sure if they are error messages~~
```
type=SERVICE_START msg=audit(1653279088.033:278): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=iptsd comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'UID="root" AUID="unset"
type=SYSTEM_RUNLEVEL msg=audit(1653279088.039:279): pid=3452 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='old-level=N new-level=5 comm="systemd-update-utmp" exe="/usr/lib/systemd/systemd-update-utmp" hostname=? addr=? terminal=? res=success'UID="root" AUID="unset"
type=SERVICE_START msg=audit(1653279088.058:280): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-update-utmp-runlevel comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'UID="root" AUID="unset"
type=SERVICE_STOP msg=audit(1653279088.058:281): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-update-utmp-runlevel comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'UID="root" AUID="unset"
type=SERVICE_STOP msg=audit(1653279088.069:282): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=iptsd comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'UID="root" AUID="unset"
```

**Comment by StollD** [comment]
You need to fix the SELinux contexts: https://github.com/quo/iptsd/issues/1#issuecomment-1198511909

An alternative would be to build iptsd into an rpm file by running `rpkg local` in the repository root and installing the rpm file it produces.

**Comment by swinzy** [comment]
> You need to fix the SELinux contexts: [#1 (comment)](https://github.com/quo/iptsd/issues/1#issuecomment-1198511909)
> 
> An alternative would be to build iptsd into an rpm file by running `rpkg local` in the repository root and installing the rpm file it produces.

Thanks for the information. I've assigned correct SELinux labels to the files and now it runs perfectly. Appreciates!
